packages/fluentui/react-northstar/src/components/Datepicker/DatepickerCalendar.tsx

```diff
@@ -295,90 +302,107 @@ export const DatepickerCalendar: ComponentWithAs<'div', DatepickerCalendarProps>
     }
   }, [visibleGrid, shouldFocusInDayGrid]);
 
-  const renderWeekRow = (week: IDay[]) =>
-    _.map(week, (day: IDay) =>
-      createShorthand(DatepickerCalendarCell, calendarCell, {
-        defaultProps: () =>
-          getA11yProps('calendarCell', {
-            content: day.date,
-            key: day.key,
-            'aria-label': format(
-              props.calendarCellFormatString,
-              formatMonthDayYear(day.originalDate, dateFormatting),
-              days[day.originalDate.getDay()],
-            ),
-            selected: day.isSelected,
-            disabled: !day.isInBounds,
-            quiet: !day.isInMonth,
-            today: compareDates(day.originalDate, props.today ?? new Date()),
-            ref: compareDates(gridNavigatedDate, day.originalDate) ? focusDateRef : null,
-            onFocus: () => setGridNavigatedDate(day.originalDate),
-          }),
-        overrideProps: (predefinedProps: DatepickerCalendarCellProps): DatepickerCalendarCellProps => ({
-          onClick: e => {
-            _.invoke(props, 'onDateChange', e, { ...props, value: day });
-            _.invoke(predefinedProps, 'onClick', e, { ...predefinedProps, value: day });
-          },
+  const renderCell = (day: IDay, content) =>
+    createShorthand(DatepickerCalendarCell, calendarCell, {
+      defaultProps: () =>
+        getA11yProps('calendarCell', {
+          content,
+          key: day.key,
+          selected: day.isSelected,
+          disabled: !day.isInBounds,
+          quiet: !day.isInMonth,
+          today: compareDates(day.originalDate, props.today ?? new Date()),
+        }),
+    });
+
+  const renderCellButton = (day: IDay) =>
+    createShorthand(DatepickerCalendarCellButton, calendarCellButton, {
+      defaultProps: () =>
+        getA11yProps('calendarCell', {
+          content: day.date,
+          'aria-label': formatMonthDayYear(day.originalDate, dateFormatting),
+          selected: day.isSelected,
+          disabled: !day.isInBounds,
+          quiet: !day.isInMonth,
+          today: compareDates(day.originalDate, props.today ?? new Date()),
+          ref: compareDates(gridNavigatedDate, day.originalDate) ? focusDateRef : null,
+          onFocus: () => setGridNavigatedDate(day.originalDate),
```

Comment:
```suggestion
```

This should go to `overrideProps`

Reply:
Does it make sense to override props that are intended to be used per day (both ref and onFocus); though the user does not receive the 'day' information in the exposed interface?

Reply:
For `onFocus` you have to `predefinedProps.onFocus`, for `ref` it's the same case as linked in a previous comment

Reply:
Updated onFocus. 

Kept the ref. I wanted to ask: should I create a ref prop in the DatepickerCalendarCellButton?

Reply:
> I create a ref prop in the DatepickerCalendarCellButton

No, `ref` is not a React prop and it should be in props, if you have typings issue, please fix it in exact usage, something like `DatepickerCalendarCellButtonProps & { ref: A_TYPE_THAT_CORRESPONDS }`

Reply:
Thanks. Fully updated.